extensions/vscode/src/state.test.ts

```diff
@@ -0,0 +1,376 @@
+
+      // Two API calls were triggered, each for every different
+      expect(mockClient.configurations.get).toHaveBeenCalledTimes(2);
+      expect(currentConfig).toEqual(secondConfig);
+      expect(publisherState.configurations).toEqual([config, secondConfig]);
+    });
+
+    test("error responses from API", async () => {
+      const contentRecordState: DeploymentSelectorState =
+        selectionStateFactory.build();
+
+      const { mockContext } = mkExtensionContextStateMock({});
+      const publisherState = new PublisherState(mockContext);
+
+      // setup existing content record in cache
+      const contentRecord = preContentRecordFactory.build({
+        deploymentPath: contentRecordState.deploymentPath,
+      });
+      publisherState.contentRecords.push(contentRecord);
+
+      // setup fake 404 error from api client
+      const axiosErr = new AxiosError();
+      axiosErr.response = {
+        data: "",
+        status: 404,
+        statusText: "404",
+        headers: {},
+        config: { headers: new AxiosHeaders() },
+      };
+      mockClient.configurations.get.mockRejectedValue(axiosErr);
+
+      // set an initial state so it tries to pull from API
+      await publisherState.updateSelection(contentRecordState);
+
+      let currentConfig = await publisherState.getSelectedConfiguration();
+      expect(mockClient.configurations.get).toHaveBeenCalledTimes(1);
+      expect(mockClient.configurations.get).toHaveBeenCalledWith(
+        contentRecord.configurationName,
+        contentRecord.projectDir,
+      );
+
+      // 404 errors are just ignored
+      expect(currentConfig).toEqual(undefined);
+      expect(publisherState.configurations).toEqual([]);
+      expect(window.showInformationMessage).not.toHaveBeenCalled();
+
+      // NOT 404 errors are shown
+      axiosErr.response = {
+        data: "custom test error",
+        status: 401,
+        statusText: "401",
+        headers: {},
+        config: { headers: new AxiosHeaders() },
+      };
+      mockClient.contentRecords.get.mockRejectedValue(axiosErr);
+
+      currentConfig = await publisherState.getSelectedConfiguration();
+      expect(mockClient.configurations.get).toHaveBeenCalledTimes(2);
+      expect(mockClient.configurations.get).toHaveBeenCalledWith(
+        contentRecord.configurationName,
+        contentRecord.projectDir,
+      );
+
+      // This error is propagated up now
+      expect(currentConfig).toEqual(undefined);
+      expect(publisherState.configurations).toEqual([]);
+      expect(window.showInformationMessage).toHaveBeenCalledWith(
+        "Unable to retrieve deployment configuration: custom test error",
+      );
+    });
+  });
+
+  test.todo("getSelectedConfiguration", () => {});
+
+  test.todo("refreshContentRecords", () => {});
+
+  test.todo("refreshConfigurations", () => {});
+
+  test.todo("validConfigs", () => {});
+
+  test.todo("configsInError", () => {});
+
+  test.todo("refreshCredentials", () => {});
```

Comment:
I'll keep working on these, to be landing in future PRs, but wanted to push and share the current progress.